### PR TITLE
Adds an option to fetch and log warnings

### LINKED ISSUE #133: Fetch and log MySQL warnings
It is already possible to detect when warnings occur in a query, using waring_count in the ok_packet record.

``` Erlang
Result = emysql:execute(test_pool,
                        <<"DROP TABLE IF EXISTS not_a_table">>),
#ok_packet{warning_count = W} = Result.
```

But then, if you want to fetch the warnings by executing a SHOW WARNINGS (in a connection pool with more than one connection), you cannot guarrantee that this query is executed on the same connection as the previous query which caused the warnings.

To solve this and to make warnings more visible to the programmer, I propose a config flag `{warnings, true}` for `emysql:add_pool/2` (i.e. add it to the `#pool{}` and `#emysql_connection{}` records) that would cause warnings to be fetched automatically and logged in the standard OTP way, using `error_logger:warning_report/1` (http://www.erlang.org/doc/man/error_logger.html#warning_report-1), which by default makes it appear just like any other Erlang error in the shell:

```
=ERROR REPORT==== 25-Mar-2014::16:43:23 ===
{emysql_warnings,[<<"Unknown table 'hello_database.not_a_table'">>]}
```

The warnings flag could be false by default to not change any of the current behaviour.

### PR DESCRIPTION
**Pull request by zuiderkwast**
Fixes #133.

emysql:add_pool/2 accepts a {warnings, boolean()} in the config proplist. It is false by default.

I have added a test that causes a MySQL warning. It asserts that there is exactly one warning. It doesn't really assert that the warning is being logged though.

### CODE REVIEW

**Comment by jlouis** [comment]
I like the patch very much. Does it pass the dialyzer? I'll add some comments to the patch itself.

File: src/emysql_conn.erl [hunk 1]
```diff
@@ -90,11 +90,16 @@ execute(Connection, StmtName, []) when is_atom(StmtName) ->
     prepare_statement(Connection, StmtName),
     StmtNameBin = atom_to_binary(StmtName, utf8),
     Packet = <<?COM_QUERY, "EXECUTE ", StmtNameBin/binary>>,
-    emysql_tcp:send_and_recv_packet(Connection#emysql_connection.socket, Packet, 0);
-execute(#emysql_connection { socket = Sock }, Query, []) ->
+    Ret = emysql_tcp:send_and_recv_packet(Connection#emysql_connection.socket, Packet, 0),
+    Connection#emysql_connection.warnings andalso
+	log_warnings(Connection#emysql_connection.socket, Ret),
+    Ret;
+execute(#emysql_connection { socket = Sock, warnings = Warnings }, Query, []) ->
```
Comment (jlouis) [inline src/emysql_conn.erl]: The above idiom is repeated all over the file. I suggest writing a helper function which handles this, just by writing a local `send_recv` function and then pass `send_recv(Connection, Packet, 0)` and unpack `Connection` inside the function. this way we can probably fold the general scheme into one function.

File: src/emysql_conn.erl [hunk 2]
```diff
@@ -129,6 +140,8 @@ prepare(Connection, Name, Statement) ->
     Packet = <<?COM_QUERY, "PREPARE ", (list_to_binary(Name))/binary, " FROM ", StatementBin/binary>>,  % todo: utf8?
     case emysql_tcp:send_and_recv_packet(Connection#emysql_connection.socket, Packet, 0) of
         OK when is_record(OK, ok_packet) ->
+            Connection#emysql_connection.warnings andalso
+                log_warnings(Connection#emysql_connection.socket, OK),
```
Comment (jlouis) [inline src/emysql_conn.erl]: Fold this into the general function as well if possible.

File: src/emysql_conn.erl (hunk 1 above)
Comment (zuiderkwast) [inline src/emysql_conn.erl]: Good idea. I'll do that today. Yes, it does pass thet dialyzer.

**Comment by zuiderkwast** [comment]
I added a line of documentation. Can't `make docs` though because the docs are kind of broken. Some other function is added between the docs for add_pool/2 and the implementation of add_pool/2. (Needs to be resolved but not in this pull request.)

It does pass the tests (make test) and the dialyzer (make dialyzer).

**Comment by jlouis** [comment]
Awesome work! I'll probably add "make docs" to the make build so I can catch errors due to that with the Travis CI tool.

**Pull request merged**